File: climbing-stairs/climbing-stairs.rb
# @param {Integer} n
# @return {Integer}

def fib(n,memo)
    if memo.include?(n)
        return memo[n]
    elsif n <= 1
        return 1
    else
        memo[n] = fib(n-1,memo) + fib(n-2,memo)
        return memo[n]
    end
end

def climb_stairs(n)
        memo = Hash.new(n+1)
    return fib(n,memo)
end

File: climbing-stairs/climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    const table = new Array(n + 1);
    
    table[0] = 1;
    table[1] = 1;
    
    for (let i = 2; i <= n; i++) {
        table[i] = table[i - 1] + table[i - 2]
    }
    return table[n]
};